File: src/NocWMS-Client/src/app/models/product.ts
export class Product {
    constructor(product?: Partial<Product>) {
        if (product) {
            this.id = product.id;
            this.categoryId = product.categoryId;
            this.productName = product.productName;
            this.sellPrice = product.sellPrice;
            this.spec = product.spec;
            this.unit = product.unit;
            this.model = product.model;
        }
    }

    public id: number;
    public categoryId: number;
    public productName: string;
    public unit: string;
    public spec: string;
    public model: string;
    public sellPrice: number;
}

File: src/NocWMS-Server/Server.ApiService/Models/Product.cs
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;

namespace Server.ApiService.Models
{
    public class Product : IEntity<int>
    {
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public string ProductName { get; set; }
        public string Unit { get; set; }
        public string Spec { get; set; }
        public string Model { get; set; }
        public float SellPrice { get; set; }

        public List<Inventory> Inventories { get; set; }
        public List<InboundReceiptItem> InboundReceiptItems { get; set; }
        public List<OutboundReceiptItem> OutboundReceiptItems { get; set; }
        public List<VendorProduct> VendorProducts { get; set; }
        public Category Category { get; set; }
        public List<InventoryCell> InventoryCells { get; set; }
    }
}
